client/state/data-layer/wpcom/sites/simple-payments/index.js

```diff
@@ -0,0 +1,167 @@
+ * @param { Object } product action with product payload
+ * @returns { Object } custom post type data
+ */
+function productToCustomPost( product ) {
+	return Object.keys( product ).reduce(
+		function( payload, current ) {
+			if ( metadataSchema[ current ] ) {
+				payload.metadata.push( {
+					key: metadataSchema[ current ].metaKey,
+					value: product[ current ]
+				} );
+			}
+			return payload;
+		},
+		{
+			type: 'jp_pay_product',
+			metadata: [],
+			title: product.title,
+			content: product.description,
+		}
+	);
+}
+
+/**
+ * Issues an API request to fetch products for a site.
+ *
+ * @param  {Object}  store  Redux store
+ * @param  {Object}  action Action object
+ */
+export function requestSimplePaymentsProducts( { dispatch }, action ) {
+	const { siteId } = action;
+
+	dispatch( http( {
+		method: 'GET',
+		path: `/sites/${ siteId }/posts`,
+		query: {
+			type: 'jp_pay_product',
+			status: 'publish'
+		},
+	}, action ) );
+}
+
+/**
+ * Issues an API request to add a new product
+ * @param {Object} store Redux store
+ * @param {Object} action Action object
+ */
+export function requestSimplePaymentsProductAdd( { dispatch }, action ) {
+	const { siteId, product } = action;
+
+	dispatch( http( {
+		method: 'POST',
+		path: `/sites/${ siteId }/posts/new`,
+		body: productToCustomPost( product ),
+	}, action ) );
+}
+
+/**
+ * Issues an API request to edit a product
+ * @param {Object} store Redux store
+ * @param {Object} action Action object
+ */
+export function requestSimplePaymentsProductEdit( { dispatch }, action ) {
+	const { siteId, product, productId } = action;
+
+	dispatch( http( {
+		method: 'POST',
+		path: `/sites/${ siteId }/posts/${ productId }`,
+		body: productToCustomPost( product ),
+	}, action ) );
+}
+
+/**
+ * Issues an API request to delete a product
+ * @param {Object} store Redux store
+ * @param {Object} action Action object
+ */
+export function requestSimplePaymentsProductDelete( { dispatch }, action ) {
+	const { siteId, productId } = action;
+
+	dispatch( http( {
+		method: 'POST',
+		path: `/sites/${ siteId }/posts/${ productId }/delete`,
+	}, action ) );
+}
+
+export const addProduct = ( { dispatch }, { siteId }, next, newProduct ) =>
+	dispatch( receiveUpdateProduct( siteId, customPostToProduct( newProduct ) ) );
+
+export const deleteProduct = ( { dispatch }, { siteId }, next, deletedProduct ) =>
+	dispatch( receiveDeleteProduct( siteId, deletedProduct.ID ) );
+
+export const listProducts = ( { dispatch }, { siteId }, next, { found: numOfProducts, posts: products } ) =>
+	dispatch( receiveProductsList( siteId, numOfProducts, products.map( customPostToProduct ) ) );
+
+const announceListingProductsFailure = ( { dispatch, getState }, { siteId } ) => {
+	const site = getRawSite( getState(), siteId );
+	const error = site && site.name
+		? translate( 'Failed to retrieve products for site “%(siteName)s.”', { args: { siteName: site.name } } )
```

Comment:
Hi! I've found a possible matching string that has already been translated **18** times:
`translate( 'Failed to retrieve comments for site “%(siteName)s”' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).